cpp/include/functions.h

```diff
@@ -85,7 +85,7 @@ gdf_error gdf_add_adj_list(gdf_graph *graph);
  */
 /* ----------------------------------------------------------------------------*/
 
-gdf_error gdf_add_transpose(gdf_graph *graph);
+gdf_error gdf_add_transposed_adj_list(gdf_graph *graph);
```

Comment:
Question... would this be better named gdf_add_transposed_from_adj_list (same comment applies to all)?  My first "read" of this function name suggests that it is adding an adj_list... which is in fact the opposite of what it is doing.

Reply:
By the same token, if we go the route I suggest should gdf_add_edge_list be renamed to gdf_add_edge_list_from_adj_list?

Reply:
Thank you very much for raising this issue. I had similar doubts. So this is related to a bigger API design issue.

First there are naming inconsistencies between python functions and C++/CUDA functions; e.g. add_edge_list in python add_edge_list to the graph from a user provided edge list but gdf_add_edge_list does conversion. And add_edge_list calls C++/CUDA gdf_edge_list_view (not gdf_edge_list).

Second some functions may surprise users; e.g. view_edge_list, view_adj_list, and num_vertices can (implicitly) involve graph conversion.

So we may need to review APIs related to class Graph and C++/CUDA counter part, but this affects multiple parts, I guess we need to solicit options from the entire group. I will submit a separate issue to discuss this.